20240129-satellite-managed-encryption.md

```diff
@@ -0,0 +1,188 @@
+---
+
+# Satellite-Managed Encryption Passphrases
+
+## Essentials
+
+### Header
+
+Date: 2024-01-29
+
+Owner: mobyvb
+
+Accountable:
+- Console squad
+
+Consulted:
+- Product
+- Delivery (TODO)
+- Satellite (TODO)
+
+Informed:
+- Engineering
+
+### Context
+
+One common issue for new users is the added step of managing an encryption passphrase for a project. The user must store the encryption passphrase, and enter it when generating access credentials or using the object browser in the Satellite UI.
+
+Having a user-managed encryption passphrase has privacy benefits for the user, but it adds an additional layer of complexity to user experience. We want to continue to support users who want to manage their own encryption credentials, while allowing other users who are not interested in managing their own encryption credentials to easily use the product. 
+
+Additionally, by default, object paths are encrypted. This has privacy benefits, but prevents certain features that some users are interested in, like lexicographic ordering of object keys. Like with user-managed encryption passphrases, we want to continue supporting path encryption for users who are interested, while allowing other users who want lexicographic ordering features to create a project which has path encryption turned off.
+
+### Goals
+
+Users are provided the option to create a project that uses "satellite-managed encryption". One of these projects:
+* uses a securely-stored, random passphrase stored on the satellite as the project's encryption passphrase
+* restricts all files to be uploaded without path encryption
+
+The main goals are to:
+* provide users with an experience that does not require managing and entering an encryption passphrase in the Satellite UI
+* provide users with an easy way to view and list files in lexographical order
+
+### Approach / Design
+
+#### Key Management Store
+
+We need a secure way to store and look up project encryption passphrases on the Satellite. Someone with normal Satellite DB access should not be able to view this data, and ideally it is encrypted. At the same time, it must be very reliable, as losing data about project encryption passphrases means that data may have been lost. The primary features and concerns we should keep in mind:
+* as simple as possible from an implementation perspective, e.g. key-value lookup for project ID -> encryption passphrase
+* secure - should be encrypted, and highly restricted who/what has access to it
+* reliable - should have the same standards for uptime/data loss as metainfo DB
+
+The KMS will be made up of 3 components:
+* The master key and store - The master key will be used to encrypt and decrypt passphrases. The store will be used to securely store the master key. The store will be [Google's Secret Manager](https://cloud.google.com/security/products/secret-manager), which is a secure store for secrets such as API keys and other sensitive data.
```

Comment:
I agree that it should be easy to have this feature enabled even without an integration with a third party like Google Secret Manager. We can update the document to be more explicit about this, but I think it will naturally happen anyway because we will want to easily test this feature in storj-up without a lot of additional configuration